### ISSUE TITLE
UnusedImports: False negative when importing class in same package

### ISSUE THREAD

**Issue by marianosimone** [arturbosch/detekt#985]
For the following files

```
package test

import test.A

val a = A()
```

```
package test

class A
```

## Expected Behavior
The `import test.A` should be reported, as both files share the same package

## Current Behavior
Nothing is reported

## Your Environment
* Version used: 1.0.0-RC7-3

**Comment by marianosimone** [comment]
I do have a working solution for this, but I'd like to do some refactoring on the rule before introducing my change (specially around style and use of mutable classes). Would it be OK to submit a PR for the refactor first?

**Comment by arturbosch** [comment]
go for it :)
